test/test_bar.py

```diff
@@ -157,3 +161,27 @@ def test_bar_graphic(fake_writer):
     file_name, content = fake_writer.call_args[0]
     eq_("render.html", file_name)
     assert_in("graphic", content)
+
+
+def test_bar_render_nteract():
+    CurrentConfig.NOTEBOOK_TYPE = NotebookType.NTERACT
+    c = (
+        Bar()
+        .add_xaxis(["A", "B", "C"])
+        .add_yaxis("series0", [1, 2, 4])
+        .add_yaxis("series1", [2, 3, 6])
+    )
+    nteract_code = c.render_notebook()
+    eq_(isinstance(nteract_code, HTML), True)
+
+
+def test_bar_render_zeppelin():
+    CurrentConfig.NOTEBOOK_TYPE = NotebookType.ZEPPELIN
+    c = (
+        Bar()
+        .add_xaxis(["A", "B", "C"])
+        .add_yaxis("series0", [1, 2, 4])
+        .add_yaxis("series1", [2, 3, 6])
+    )
+    zeppelin_code = c.render_notebook()
+    eq_(zeppelin_code, None)

```

Comment:
This assertion should intercept console output and check if %html is present. It is a bit difficult but not impossible. Google online or checkout my monban project where there exist some examples.

Reply:
I know How to block it, not difficult.

* First: Create a redirector
```python
import sys


class ConsoleOutputRedirect:
    """ Wrapper to redirect stdout or stderr """

    def __init__(self, fp):
        self.fp = fp

    def write(self, s):
        self.fp.write(s)

    def writelines(self, lines):
        self.fp.writelines(lines)

    def flush(self):
        self.fp.flush()


stdout_redirect = ConsoleOutputRedirect(sys.stdout)
```

* Second: Use StringIO to keep the sys.stdout
```python
def test_bar_render_zeppelin():
    CurrentConfig.NOTEBOOK_TYPE = NotebookType.ZEPPELIN
    c = (
        Bar()
        .add_xaxis(["A", "B", "C"])
        .add_yaxis("series0", [1, 2, 4])
        .add_yaxis("series1", [2, 3, 6])
    )
    # Block Console stdout
    output_buffer = StringIO()
    stdout_redirect.fp = output_buffer
    temp_stdout = sys.stdout
    sys.stdout = stdout_redirect

    # render
    c.render_notebook()
    sys.stdout = temp_stdout

    # Block Result
    assert_in("%html", output_buffer.getvalue())
```

Reply:
Let me share some examples offline. 

I am not trying to block your PR. Just want to point out that the test thing could be done better. At the end of the day, it is the coding experience benefiting us all. We are not paid through the work.